cumulus_library/databases.py

```diff
@@ -497,3 +499,47 @@ def create_db_backend(args: dict[str, str]) -> DatabaseBackend:
         raise ValueError(f"Unexpected --db-type value '{db_type}'")
 
     return backend
+
+
+def upload_file(
+    *,
+    cursor: DatabaseCursor,
+    file: pathlib.Path,
+    study: str,
+    topic: str,
+    remote_filename: str | None = None,
+) -> str | None:
+    if db_config.db_type == "athena":
+        # We'll investigate the cursor to get the relevant params for S3 upload
+        # TODO: this could be retrieved from a config object passed to builders
+        wg_conf = cursor.connection._client.get_work_group(
+            WorkGroup=cursor.connection.work_group
+        )["WorkGroup"]["Configuration"]["ResultConfiguration"]
+        s3_path = wg_conf["OutputLocation"]
+        bucket = "/".join(s3_path.split("/")[2:3])
+        encryption_type = wg_conf.get("EncryptionConfiguration", {}).get(
+            "EncryptionOption", {}
+        )
+        if encryption_type != "SSE_KMS":
+            raise errors.AWSError(
+                f"Bucket {bucket} has unexpected encryption type {encryption_type}."
+                "AWS KMS encryption is expected for Cumulus buckets"
+            )
+        kms_arn = wg_conf.get("EncryptionConfiguration", {}).get("KmsKey", None)
+        profile = cursor.connection.profile_name
+
+        if not remote_filename:
+            remote_filename = file
+        session = boto3.Session(profile_name=profile)
+        s3_client = session.client("s3")
+        with open(file, "rb") as b_file:
+            s3_client.put_object(
+                Bucket=bucket,
+                Key=f"user_uploads/{study}/{topic}/{remote_filename}",
```

Comment:
This makes me lightly nervous - we're attempting to own `/user_uploads` on the bucket - that might conflict with whatever bucket setup they have and requires granting upload rights to a whole new space. And prevents admin-level namespacing.

For example, we at BCH have several different stacks set up (cerner, i2b2, test stacks) - this would currently share the same folder for all those.

Can we put `user_uploads` as a subfolder of `OutputLocation`? That way the user permission grant is very close as it was before (just add a little S3 permission on top, but no path changes), and admins can namespace as they like.

Reply:
i was lightly nervous about putting it in the `results` folder, since that's technically owned by athena and i suppose there's nothing to stop athena from wiping that folder out? if we're sure that would be safe, that would be fine.

Reply:
ah what a good point!

Reply:
I think this issue was the only thing that made me suss in this PR - I'll approve to get out of you way if we're namespacing this.

Reply:
For extra future proofing, should we name the subfolder `cumulus` or something? That way we could throw things that aren't user uploads in there at some point too, if we find a need. Don't feel strongly about this, and we could always take a different name in there at that point.

Reply:
yeah, i'm going to change it to `cumulus_user_uploads`, and I am also going to add the db name to this path.